### ISSUE TITLE
There's an option ignored by the story to stealthily discover the Alpha ship in Hope without being detected

### ISSUE THREAD

**Issue by Pointedstick** [endless-sky/endless-sky#1647]
At that point in your life, your flagship probably has a cloaking device. If you bring a single ship, stuff it to the gills with Wanderer Ramscoops and fly cloaked the entire time, then it's quite simple to discover the Giftbringer undetected. This is what I did, in fact, because the Korath ships were way too tough for me (I defeated the Giftbringer primarily with Thunderheads).

Later on in the mission, Danforth mentions that if you had discovered the Giftbringer and then returned for reinforcements, the Alphas would have been alerted to your presence and fled. But not if you were cloaked when you discovered them and left without fighting! Seems like a missed opportunity: if you had the option to discover the ship while cloaked and return for reinforcements without being seen, then you could bring the planetary invasion force and destroy the occupied base right after fighting the Giftbringer in orbit.

**Comment by Bladewood** [comment]
It's important to note that particle effects are not sampled in red like cloaked ships, implying that effects like the ones made by jump drives are not cloaked, and so the Alphas would be able to detect your entrance anyway.
And what I just said should be changed, because of things like the take-out-the-alpha-nukes mission.

**Comment by Pointedstick** [comment]
Yeah, it's heavily implied that you're totally invisible while cloaked, barring some yet-to-be-announced super cloaked-ship detector technology that the Quarg probably have.

**Comment by Bladewood** [comment]
You're not totally invisible, according to Alastair Korban. You are vulnerable to gravitational sensors.

**Comment by KiLEdEnNis** [comment]
in the title, "Hope" you mean Host?

**Comment by Pointedstick** [comment]
Er, yes, Host.

**Comment by Amazinite** [comment]
It's not currently possible for a mission to determine if the player is cloaked, only that they have a cloak, and having a cloak doesn't guarantee that you're using it. There's also the possibility of the player having escorts which aren't cloaked, so I'm really not sure how a mission could reasonably check all that it would need to check to have being cloaked have an impact on the story.

Not to say that I don't like the idea of the game's story being more responsive to your action, because I like this idea, but I don't see it being something we implement any time soon. This issue might need further consideration to decide whether it's worth the effort to implement a whole new mechanic for missions checking if you 1) enter a system cloaked, 2) remain cloaked, and 3) land while cloaked. Perhaps something like #5248 could do, but instead of triggering actions based off of an NPC's status change, we do so for the player's ships. The mission syntax for that would need to be discussed, though. Perhaps something like this:
```
mission "Land Cloaked"
	...
	flagship # Observe the actions of your flagship.
		on uncloaked # Triggers if the ship is not cloaked.
			system "Sol" # Location filter akin to on enter, so that the
				     # on uncloaked only triggers in the correct system.
			set "wasn't cloaked"
	escorts # Observe the actions of all your escorts.
		on uncloaked
			system "Sol"
			set "wasn't cloaked"
	...
	on complete
		conversation
			branch uncloaked
				has "wasn't cloaked"
			`	You sneak onto the planet unseen.`
			...
			
			label uncloaked
			`	They saw you coming!`
			...
```

**Comment by warp-core** [comment]
What if it were possible to base this off of if the player or any if their ships ever became targetable to the npc?

**Comment by Hecter94** [comment]
Could you do this with a timer ship? Have an invisible ship in the system hostile to the player but friendly to the alphas which will target and shoot at anything it's able to see, killing itself in the process.
If it ever spots the player or any of their escorts, it will fire and die, and fail a mission and then that condition can be checked for with any followup missions and referenced.

If we think that would work, I can take a crack at this.

I respect that is a hacky solution though, so if we'd rather just wait until we have proper syntax to support this kind of thing before making a mission like this, I understand that as well.

**Comment by TheGiraffe3** [comment]
We have `on encounter` at this point. It looks like solving this issue would require a refactor of the missions directly involving the Alphas, though.
